compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -448,42 +288,20 @@ public STNode resumeParsing(ParserRuleContext context, Object... args) {
                 return parseLeftArrowToken();
             case RECEIVE_WORKERS:
                 return parseReceiveWorkers();
-            case WAIT_KEYWORD:
-                return parseWaitKeyword();
-            case WAIT_FUTURE_EXPR_END:
-                return parseWaitFutureExprEnd((int) args[0]);
             case WAIT_FIELD_NAME:// revisit
                 return parseWaitField();
```

Comment:
we can check if this context is required in resumeParsing and remove `// revisit` comment